### PR TITLE
[camel-k] Added test for disabled API client

### PR DESCRIPTION
**Pull request by mmuzikar**
//skip-ci

### CODE REVIEW

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: ui-tests/src/test/resources/features/other/camelk.feature [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#@sustainer: mmuzikar@redhat.com
```
Comment (avano) [inline ui-tests/src/test/resources/features/other/camelk.feature]: I don't know if it is necessary, but add a space between `#` and `@`

File: ui-tests/src/test/resources/features/other/camelk.feature [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+@camel-k
+Feature: Camel-k Runtime
+
+  Background:
+    And clean application state
+    And change runtime to camelk
+    Given log into the Syndesis
+
+  @camel-k-api-client
```
Comment (avano) [inline ui-tests/src/test/resources/features/other/camelk.feature]: add jira annotation to the related issue

File: utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java [hunk 3]
```diff
@@ -84,19 +89,81 @@ public static void waitForSyndesis() {
...
+                deployCamelK();
+                waitForCamelK();
+            }
+            Syndesis syndesis = ResourceFactory.get(Syndesis.class);
+            Map<String, Object> cr = syndesis.getDeployedCr();
+            Map<String, Object> spec = (Map<String, Object>) cr.get("spec");
+            Map<String, Object> addons = (Map<String, Object>) spec.get("addons");
+            Map<String, Object> camelk = (Map<String, Object>) addons.get("camelk");
+
+            if (!camelk.containsKey("enabled") || !((boolean) camelk.get("enabled"))) {
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java]: you can use `Syndesis#updateAddon()` for enabling/disabling an addon

File: utilities/src/main/java/io/syndesis/qe/endpoints/TestSupport.java [hunk 4]
```diff
@@ -57,6 +58,11 @@ public void resetDB() {
                         1, 5 * 60,
                         "Some integration was not deleted successfully in time. Integration name: " +
                             OpenShiftUtils.getPodByPartialName("i-").get().getMetadata().getName());
+                } else {
+                    ResourceFactory.get(CamelK.class).resetState();
+                    if (!OpenShiftUtils.getPodByPartialName("camel-k-operator").isPresent()){
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/endpoints/TestSupport.java]: you can't fool me with `//skip-ci` :grin: missing space between `)){`

File: utilities/src/main/java/io/syndesis/qe/resource/impl/CamelK.java [hunk 5]
```diff
@@ -55,8 +61,9 @@ public void undeploy() {
         if (OpenShiftUtils.getAnyPod("syndesis.io/component", "syndesis-server").isPresent()) {
             // It is needed to reset-db first, otherwise server would create the integrations again
             TestSupport.getInstance().resetDB();
-            removeViaBinary();
+            resetState();
             OpenShiftUtils.getInstance().apps().deployments().withName("camel-k-operator").delete();
+            OpenShiftUtils.getPodByPartialName("camel-k-operator").ifPresent(pod -> OpenShiftUtils.getInstance().deletePod(pod));
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/resource/impl/CamelK.java]: less hacky would be selecting by label `camel.apache.org/component: operator`

File: utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java [hunk 6]
```diff
@@ -84,19 +89,81 @@ public static void waitForSyndesis() {
...
+                camelk.put("enabled", true);
+                try {
+                    syndesis.editCr(cr);
+                } catch (IOException e) {
+                    fail("There was an error while updating the CR", e);
+                }
+                needsReload = true;
+            }
+        } else {
+            if (ResourceFactory.get(CamelK.class).isReady()){
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java]: `()){`

File: utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java [hunk 7]
```diff
@@ -67,6 +71,7 @@ public void cleanBuilds() {
     @When("^deploy Syndesis$")
     public static void deploySyndesis() {
         ResourceFactory.create(Syndesis.class);
+        changeRuntime(TestConfiguration.syndesisRuntime());
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java]: wouldn't it make sense to include it in `Syndesis#deploy()`?

**Comment by mmuzikar** [comment]
wtf

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java [hunk 8]
```diff
@@ -84,19 +86,60 @@ public static void waitForSyndesis() {
...
+    public static void waitForCamelK() {
         OpenShiftUtils.getInstance().waiters()
             .areExactlyNPodsReady(1, "camel.apache.org/component", "operator")
             .interval(TimeUnit.SECONDS, 20)
             .timeout(TimeUnit.MINUTES, 5)
             .waitFor();
     }
 
+    @When("^change runtime to (springboot|camelk)$")
+    public static void changeRuntime(String runtime) {
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java]: please move the implementation to Syndesis class

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**